internal/services/signalr/web_pubsub_custom_certificate_binding.go

```diff
@@ -0,0 +1,282 @@
+package signalr
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/webpubsub/2023-02-01/webpubsub"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	keyVaultParse "github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/parse"
+	keyVaultValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CustomCertBindingWebPubsubModel struct {
+	Name               string `tfschema:"name"`
+	WebPubsubId        string `tfschema:"web_pubsub_id"`
+	CustomCertId       string `tfschema:"custom_certificate_id"`
+	CertificateVersion string `tfschema:"certificate_version"`
+}
+
+type CustomCertBindingWebPubsubResource struct{}
+
+var _ sdk.Resource = CustomCertBindingWebPubsubResource{}
+
+func (r CustomCertBindingWebPubsubResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"web_pubsub_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: webpubsub.ValidateWebPubSubID,
+		},
+
+		"custom_certificate_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.Any(
+				keyVaultValidate.NestedItemId,
+				keyVaultValidate.NestedItemIdWithOptionalVersion,
+			),
+		},
+
+		"certificate_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r CustomCertBindingWebPubsubResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r CustomCertBindingWebPubsubResource) ModelObject() interface{} {
+	return &CustomCertBindingWebPubsubModel{}
+}
+
+func (r CustomCertBindingWebPubsubResource) ResourceType() string {
+	return "azurerm_web_pubsub_custom_certificate_binding"
```

Comment:
From looking at the portal, this is just a certificate. 

<img width="1168" alt="image" src="https://user-images.githubusercontent.com/6515518/227655631-56f93a5e-aa37-422c-afba-4c9b73fd01d2.png">

Does this need to be named `azurerm_web_pubsub_custom_certificate_binding` or would `azurerm_web_pubsub_custom_certificate` be ok?